modules/broker-auth/src/main/java/io/ballerina/messaging/broker/auth/authorization/authorizer/rdbms/RdbmsAuthorizer.java

```diff
@@ -0,0 +1,175 @@
+import io.ballerina.messaging.broker.auth.exception.BrokerAuthServerException;
+import io.ballerina.messaging.broker.common.StartupContext;
+import io.ballerina.messaging.broker.common.config.BrokerConfigProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nonnull;
+import javax.sql.DataSource;
+
+/**
+ * Class provides database based @{@link Authorizer} implementation.
+ */
+public class RdbmsAuthorizer implements Authorizer {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RdbmsAuthorizer.class);
+
+    private AuthScopeStore authScopeStore;
+
+    private AuthResourceStore authResourceStore;
+
+    private AuthProvider authProvider;
+
+    /**
+     * Cache which store user id vs  user cache entry
+     */
+    private LoadingCache<String, UserCacheEntry> userCache;
+
+    @Override
+    public void initialize(AuthProvider authProvider, StartupContext startupContext, Map<String, Object> properties)
+            throws Exception {
+
+        this.authProvider = authProvider;
+        DataSource dataSource = startupContext.getService(DataSource.class);
+        BrokerConfigProvider configProvider = startupContext.getService(BrokerConfigProvider.class);
+        BrokerAuthConfiguration brokerAuthConfiguration = configProvider.getConfigurationObject(
+                BrokerAuthConfiguration.NAMESPACE, BrokerAuthConfiguration.class);
+        authResourceStore = new AuthResourceStoreImpl(brokerAuthConfiguration, dataSource, authProvider);
+        authScopeStore = new AuthScopeStoreImpl(brokerAuthConfiguration, dataSource);
+        userCache = CacheBuilder.newBuilder()
+                                .maximumSize(brokerAuthConfiguration.getAuthorization().getCache()
+                                                                    .getSize())
+                                .expireAfterWrite(brokerAuthConfiguration.getAuthorization()
+                                                                         .getCache()
+                                                                         .getTimeout(),
+                                                  TimeUnit.MINUTES)
+                                .build(new UserCacheLoader());
+    }
+
+    @Override
+    public boolean authorize(String scopeName, String userId)
+            throws BrokerAuthException, BrokerAuthServerException, BrokerAuthNotFoundException {
+        try {
+            if (userId != null) {
+                UserCacheEntry userCacheEntry = userCache.get(userId);
+                boolean anyMatch = userCacheEntry.getAuthorizedScopes()
+                                                 .stream()
+                                                 .anyMatch(s -> s.equals(scopeName));
+                if (anyMatch) {
```

Comment:
Shall we use a variable name like `isAuthorized`?

Reply:
Fixed in 384b733